sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -197,15 +211,16 @@ internal virtual void Wait(TimeSpan time, CancellationToken cancellationToken)
 
         private bool ShouldRetryInternal(HttpMessage message, Exception? exception)
         {
+            var classifier = _responseClassifier ?? message.ResponseClassifier;
```

Comment:
IIRC, for classifier error behavior, the message classifier will take precedence over the pipeline classifier (we should verify this, though).  It looks like here, the policy classifier will always win, so can never be overridden on a per invocation basis, i.e. via request context.  Would it make sense to be consistent, or is there a reason we want the policy to always win for this case?

Reply:
So long as any `true` opts into retries, does the order matter?   If we're saying that the message would obviate the custom classifier - then that defeats my goal in asking for this.   I wanted my custom classifier to own the decision but didn't want to have to create a custom retry policy just to achieve that.

Reply:
> Would it make sense to be consistent, or is there a reason we want the policy to always win for this case?

Good question - the intent was to override the default/library classifier. I do think requestContext classifier should have highest precedence though, so I will need to fix that. Here is what I think the precedence should be from lowest to highest:

1. Default ResponseClassifier
2. Library-specified ResponseClassifier via pipeline build options
3. RetryPolicy ResponseClassifier (this one!)
4. RequestContext ResponseClassifier

For 4, the classifier will still be threaded through onto the HttpMessage, but we will need a way (probably via internal property) for the RetryPolicy to know that this is a per-request response classifier and it should take precedence.

Reply:
I guess we already have such an internal property, [Policies](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/HttpMessage.cs#L116), on HttpMessage.

Reply:
> I wanted my custom classifier to own the decision but didn't want to have to create a custom retry policy just to achieve that.

@jsquire, can you say more about this?  I'm a bit confused, because I understood this PR to be enabling to you set a custom classifier that you *only* set on the retry policy, and so would require you to create a custom retry policy.  But now I think I may be misunderstanding something fundamental?

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/37727 to track the documentation and test coverage since it seems that this is needed regardless of how we proceed with this change